src/autocomplete.ts

```diff
@@ -1,1654 +1,1691 @@
+        }, delay);
+    }
+
+    /**
+     * empty the screen reader announcement
+     */
+    emptyAnnouncement(delay: number = 2000) {
+        if (!this.srAnnouncements) {
+            return;
+        }
+        // in immediate case, do not use timer
+        if (delay === 0) {
+            this.srAnnouncements.textContent = '';
+            return;
+        }
+
+        clearTimeout(this.announcementEmptyTimer);
+        this.announcementEmptyTimer = setTimeout(() => {
+            this.srAnnouncements.textContent = '';
+        }, delay);
+    }
+
+    /**
+     * check if element is a selected element in the DOM
+     */
+    isSelectedElem(element: Element): boolean {
+        const sourceEntry = element && element[SELECTED_OPTION_PROP];
+        return this.multiple && sourceEntry && typeof sourceEntry === 'object';
+    }
+
+    /**
+     * get selected items DOM elements
+     */
+    getSelectedElems(): HTMLElement[] {
+        const result = [];
+        for (let i = 0, l = this.wrapper.childNodes.length; i < l; i += 1) {
+            const node: ChildNode = this.wrapper.childNodes[i];
+            if (node.nodeType === 1 && this.isSelectedElem(node as HTMLElement)) {
+                result.push(node as HTMLElement);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * remove object from selected options array
+     */
+    removeEntryFromSelected(entry: any) {
+        // prevent removing if deletions are disabled
+        if (this.deletionsDisabled) {
+            return;
+        }
+
+        // check for explicit match first
+        let index: number = this.selected.indexOf(entry);
+        // if object reference check did not work found, do a manual value check
+        if (index === -1) {
+            for (let i = 0, l = this.selected.length; i < l; i += 1) {
+                if (this.selected[i].value === entry.value) {
+                    index = i;
+                    break;
+                }
+            }
+        }
+        if (index > -1 && this.selected[index]) {
+            const option: any = mergeObjects(this.selected[index]);
+            const label = option.label;
+            // set option or checkbox to not be selected
+            setElementState(option.element, false, this);
+            // update array
+            this.selected.splice(index, 1);
+            // fire on delete before updating the DOM
+            this.triggerOptionCallback('onDelete', [option]);
+            // update original element values based on now selected items
+            this.setSourceElementValues();
+            // re-build the selected items markup
+            this.buildMultiSelected();
+            // update input size in autoGrow mode
+            this.triggerAutoGrow();
+            // make sure to announce deletion to screen reader users
+            this.announce(`${label} ${this.options.srDeletedText}`, 0);
+            // return focus to input
+            this.input.focus();

```

Comment:
This change will have inadvertently caused a bug in single select mode when the `confirmOnBlur` option is true. After making a selection, if the input's value is then changed, and blurred from, the `removeEntryFromSelected` method is called. So even though the user has blurred from the input, this change will mean that they're forced back to input immediately.